### ISSUE TITLE
"Stream all rows" is not at all obvious

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#1142]
Got a question about how to download all rows - the current option isn't at all clear.

<img width="668" alt="loans__ppp_loans__9_511_rows_where_where_search_matches__tech__sorted_by_rowid" src="https://user-images.githubusercontent.com/9599/101977057-ac660b00-3bff-11eb-88f4-c93ffd03d3e0.png">

**Comment by nitinpaul** [comment]
'Include all rows' seem like a fairly obvious alternative

**Comment by simonw** [comment]
Yeah, maybe with the number of rows to make it completely clear. `Include all 2,455 rows` perhaps.

**Comment by simonw** [comment]
You're actually choosing between two options here: the 100 rows you can see on the screen, or the x,000 rows that match the current query.

Maybe a radio box would be more obvious?

**Comment by nitinpaul** [comment]
Agreed, it would definitely provide better controls. However, I do feel it makes for a bit of inconsistent UX for the 'Advanced export' section, with links to download for JSON, checkboxes and radio buttons + button to download for CSV. Do you think this example makes the UX a bit nicer/consistent?

![Screenshot 2020-12-13 at 5 38 43 PM](https://user-images.githubusercontent.com/6622733/102011444-1dc1cd00-3d6a-11eb-9e38-5af198161e80.png)

I could give it a try if you'd like but I've never contributed to an actual project!

**Comment by nitinpaul** [comment]
Ok! I'll give it a shot!

**Comment by nitinpaul** [comment]
Alright I could give it a try! This might be a stupid question, can you tell me how to run the server from my fork? So that I can test the changes?

**Comment by simonw** [comment]
To check out and start the server:

    /tmp % git clone git@github.com:nitinpaul/datasette
    Cloning into 'datasette'...
    remote: Enumerating objects: 124, done.
    # ...
    datasette % python3 -m venv venv
    datasette % source venv/bin/activate
    (venv) datasette % pip install -e '.[test]'
    Obtaining file:///private/tmp/datasette
    Collecting asgiref<3.4.0,>=3.2.10
    Using cached asgiref-3.3.1-py3-none-any.whl (19 kB)
    # ...
    (venv) datasette % datasette
    INFO:     Started server process [24002]
    INFO:     Waiting for application startup.
    INFO:     Application startup complete.
    INFO:     Uvicorn running on http://127.0.0.1:8001 (Press CTRL+C to quit)

And to run the tests:

    (venv) datasette % pytest
    ======================================================================== test session starts ========================================================================
    platform darwin -- Python 3.9.1, pytest-6.1.2, py-1.10.0, pluggy-0.13.1
    SQLite: 3.34.0
    rootdir: /private/tmp/datasette, configfile: pytest.ini
    plugins: asyncio-0.14.0, timeout-1.4.2
    collected 841 items                                                                                                                                                 

    tests/test_package.py ..                                                                                                                                      [  0%]

**Comment by simonw** [comment]
I'm not sure about the radio boxes for JSON, just because you can't right-click on a radio box and copy it to your clipboard like you can with links. Worth trying it out though.

The radio boxes for that CSV option are definitely the right way to go.

**Comment by nitinpaul** [comment]
You're right, probably more straightforward to have the links for JSON. I was imagining to toggle the `href` for the 'Export JSON' link (button) to the selected shape, but it'll probably be needlessly complex in the end.
